backend/src/schema/resolvers/registration.js

```diff
@@ -0,0 +1,107 @@
+import { UserInputError } from 'apollo-server'
+import uuid from 'uuid/v4'
+import { neode } from '../../bootstrap/neo4j'
+import fileUpload from './fileUpload'
+import encryptPassword from '../../helpers/encryptPassword'
+
+const instance = neode()
+
+/*
+ * TODO: remove this function as soon type `User` has no `email` property
+ * anymore
+ */
```

Comment:
I made a refactoring to the type `User` and left it at a dangerously in-between state. So, right now, the `User` has an attribute `email`. The idea of the the signup is, however, that you add an `EmailAddress` *without* a user account and then create the user account and connect it with the orphaned email address. This `email` attribute is a left-over. It should be completely removed at some point. But I think it's OK for now to keep it, it doesn't break anything and since we're free to create our own resolvers, we can also have a `User.email` resolver, which basically just `MATCH(u:User)-[:PRIMARY_EMAIL]->(e:EmailAddress)` pulls in the email. This is also a preparation for having multiple email addresses.

Reply:
great, thanks for the explanation